parts/linux/cloud-init/artifacts/components.json

```diff
@@ -8,512 +10,18 @@
         "1.8.22"
       ]
     },
+
+    //new
     {
-      "downloadURL": "mcr.microsoft.com/oss/kubernetes/metrics-server:*",
+      "name": "nvidia-cuda",
+      "type": "containerimages //extend to OCI |Tar|tgz when merging with packages,
+      "action": "cacheandfetch|fetchonly|config",  //where cacheandfetch is cached at build time and also to fetch at provisioning time if not cached. fetchonly is only fetching at provisioning time but not caching. config is neither cache nor fetch but just for storing data for other purpose. Open to change the names of the values.
+    "registryURL": "mcr.microsoft.com/",
```

Comment:
could we have different fields for different `type`s? e.g. I think for type `image` it would make more sense to have `registryURL`+ `repo`, and for type `tar` we have `downloadURL`, etc.

Reply:
Right I also thought about this.
Or keep separating `ContainerImages` and `Packages` into 2 groups at root level if they have many differences. Then `type` could be more "local". For example, each Package has a `type: OCI | tar| tgz` etc. ContainerImage has no `type`

Reply:
Do you mean we would have different sets of fields in the same json? That could be annoying eventually when we will want to parse the json in the code (e.g. for testing)

Reply:
fair, though I don't think it would be a huge deal

we could always keep container images and package separate to deal with that

Reply:
Is there a clear benefit to merging `ContainerImages` and `Packages` schema?